### ISSUE TITLE
Display legacy products as phases on project plan to managers only

### ISSUE THREAD

**Issue by vic-topcoder** [appirio-tech/connect-app#2309]
We must support legacy products when upgrading projects into a new system. The goal is to allow information transferred into the new project, but rebuild the project plan with only the new phases that support timeline template.

The solution is to display the legacy products as manager/copilot read-only phase where in the Timeline tab we'll display a message (@acshields we'll need your help on language). Something in the lines of "We transfer this phase as a read-only from a legacy project. It is visible only to managers and copilots. Please use the phase catalog to add the appropriate phase with a timeline bellow."

Legacy products should not be visible on the phase catalog.

**Comment by vic-topcoder** [comment]
Let's figure out if we need to display the legacy product as delivered to the customer when we upgrade projects - that's a question for Joey and team. @acshields can you please clarify the requirements there? Thanks

**Comment by vikasrohit** [comment]
@vic-topcoder @acshields There are couple of things here:
1. There is no option for upgrading an existing project (with project plan but no timelines) to a project with plan and timelines both. We have a migration path for a project without plan to project with project plan. So, I don't think we would have this case for project upgrades.
2. However, we can have this scenario when an existing project (with project plan but no milestones) is rendered after the 2.4.4 release. At that time, we won't show the timelines tab at all for such projects. If you like to indicate the ability to add milestones there, we can show the timelines tab with the message you specified.
For reference we made the timelines tab hidden for old projects in this issue #2308.

**Comment by vikasrohit** [comment]
@vic-topcoder do we have designs for indicating the legacy phases?

**Comment by vikasrohit** [comment]
Moving out of Community fix milestone as requirements are not yet clear.

**Comment by acshields** [comment]
@vic-topcoder @vikasrohit - Is this still where you are looking for follow-up from me? "Let's figure out if we need to display the legacy product as delivered to the customer when we upgrade projects - that's a question for Joey and team. @acshields can you please clarify the requirements there?"

**Comment by vikasrohit** [comment]
Yes, @acshields there are still unclear requirements here. To summarize the queries again:

1. Do we really want to make the phases with legacy products as read only? If we do that, all existing projects (which are using phases) would get blocked and copilots might get confused and irritated. They have to copy their existing conversations in the phase to new phase and so on. So, IMO, we should not mark the phases with legacy products as readonly. Let me know what do you think?

2. Without marking such phases as read only, we can still show a message that says `this phase is using legacy type, please create new phase with new products to leverage the new timeline and milestones.` So, 3 queries here:
  2.a Should we do this change?
  2.b If yes, what should be the copy?
  2.c If yes, what should be the design?

**NOTE:** There is no way right now to migrate existing phases without milestones to phases with milestones. One has to create a new phase to use the new feature of timeline and milestones.
